vmdb/app/models/orchestration_stack.rb

```diff
@@ -40,25 +40,39 @@ def total_cloud_networks
     cloud_networks.size
   end
 
-  # @param options [Hash] what to update for the stack. Option keys and values are:
-  #   :template (String, URI, S3::S3Object, Object) - A new stack template.
-  #     This may be provided in a number of formats including:
-  #       a String, containing the template in CFN or HOT format.
-  #       a URL String pointing to the document in S3.
-  #       a URI object pointing to the document in S3.
-  #       an S3::S3Object which contains the template.
-  #       an Object which responds to #to_json and returns the template.
-  #   :parameters (Hash) - A hash that specifies the input parameters of the new stack.
+  def self.create_stack(orchestration_manager, stack_name, template, options = {})
+    klass = "OrchestrationStack#{orchestration_manager.class.name.match(/Ems(.*)/)[1]}".constantize
+    ems_ref = klass.raw_create_stack(orchestration_manager, stack_name, template, options)
+
+    klass.create(:name                   => stack_name,
+                 :ems_ref                => ems_ref,
+                 :status                 => 'CREATE_IN_PROGRESS',
+                 :ext_management_system  => orchestration_manager,
+                 :orchestration_template => template)
+  end
+
+  def self.raw_create_stack(_orchestration_manager, _stack_name, _template, _options = {})
+    raise NotImplementedError, "raw_create_stack must be implemented in a subclass"
+  end
```

Comment:
From a pluggability perspective where plugins will eventually be separate gems, does it make sense that when two completely independent providers with code that happens to look the same that we write their implementation in the base?  When the third one comes along, it's unlikely to share the code, meaning some are in the base class and the rest aren't?  I get the comments about being DRY, but in this case, I'm not so sure.